scripts/core/EXT_Exp_CounterBasedEvents.rst

```diff
@@ -0,0 +1,50 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+%>
+:orphan:
+
+.. _ZE_experimental_event_counter_based:
+
+=========================
+ Counter-Based Events Extension
+=========================
+
+API
+----
+ 
+* Enumerations
+
+    * ${x}_event_counter_based_exp_flags_t
+    * ${x}_event_counter_based_exp_version_t
+
+* Structures
+
+    * ${x}_event_counter_based_exp_desc_t
+
+Counter-Based Events
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, events in Level Zero contains one of two states: signaled or not signaled.
+Signaling of an event, as well as resetting its state, can be done from either host or
+device.
+
+Another way of tracking the state of an event is by using a counter, which is incremented
+every time a task has completed in the device. Using a counter-based event may provide
+Level Zero driver implementations with the opportunity for both functional and performance
+optimizations.
+
+The following recommendations and restrictions apply to counter-based events:
+
+- Counter-based events can be used only with in-orders lists, whether those are regular or immediate.
+- Counter-based events must not be reset, i.e., a call to ${x}EventHostReset or ${x}CommandListAppendEventReset
+  is not allowed.
+- Counter-based events may be reused multiple times without a need for reset.
```

Comment:
This may be confusing without first explaining how they work.
Can you add a sentence that they automatically track the last signaling usage and that passing the event to new call that will signal it, automatically updates what event represents?
I would also add that passing event to be signaled on new calls, doesn't require the host or device to wait until previous signaling / dependency resolution is completed.

Reply:
@MichalMrozek : thanks. please see proposal below:

```md
- Counter-based events can be used only with in-orders lists, whether those are regular or immediate.
- Counter-based events must not be reset, i.e., a call to ${x}EventHostReset or ${x}CommandListAppendEventReset
  is not allowed.
- Counter-based events must not be signaled from host, i.e., a call to ${x}EventHostSignal is not allowed.
- Counter-based events may be reused multiple times without a need for reset.
- Counter-based events may be used on multiple command lists.
- Querying a counter-based event queries only the last saved counter value from the last command list that
  incremented it, i.e., a signaled counter-based event always represents the completion of the last call to which
  it was passed as signal event.
- Synchronizing on a counter-based event waits only the last saved counter value from the last command list
  that incremented it.
- A counter-based event may be passed as signaling event for a new append call without needing to wait for
  the signaling of the last call where it was used.
```

Reply:
ok